ykrt/src/mt.rs

```diff
@@ -629,57 +649,73 @@ impl Drop for MT {
     }
 }
 
+/// [MTThread]'s major job is to record what state in the "interpreting/tracing/executing"
+/// state-machine this thread is in. This enum contains the states.
+enum MTThreadState {
+    /// This thread is executing in the normal interpreter: it is not executing a trace or
+    /// recording a trace.
+    Interpreting,
+    /// This thread is recording a trace.
+    Tracing {
+        /// The [HotLocation] the trace will end at. For a top-level trace, this will be the same
+        /// [HotLocation] the trace started at; for a side-trace, tracing started elsewhere.
+        hl: Arc<Mutex<HotLocation>>,
+        /// What tracer is being used to record this trace? Needed for trace mapping.
+        thread_tracer: Box<dyn TraceRecorder>,
+        /// Records the content of data recorded via `yk_promote`.
+        promotions: Vec<usize>,
+    },
+    /// This thread is executing a trace. Note that the `dyn CompiledTrace` serves two different purposes:
+    ///
+    /// 1. It is needed for side traces and the like.
```

Comment:
"It is needed for trace execution, side traces, and the like"?

Reply:
It's not needed for trace execution AFAIK.

Reply:
Does it not contain the trace that is being executed?

Reply:
Ah sorry, I get it now. The trace is already being executed. Ignore this.